indexing-service/src/main/java/org/apache/druid/indexing/common/task/batch/parallel/ParallelIndexSupervisorTask.java

```diff
@@ -156,7 +158,8 @@ public class ParallelIndexSupervisorTask extends AbstractBatchIndexTask implemen
    * Only the compaction task can have a special base name.
    */
   private final String baseSubtaskSpecName;
-  private final InputSource baseInputSource;
+
+  private final AtomicReference<InputSource> inputSourceRef = new AtomicReference<>();
```

Comment:
You can still call it the `baseInputSource`. `inputSourceRef` is a confusing name.

Reply:
Done